test/integration_tests/test_bids_download.py

```diff
@@ -0,0 +1,253 @@
+    "bids_dataset/sub-02/ses-post/func/sub-02_ses-post_task-livingnonlivingdecisionwithplainormirrorreversedtext_run-02_bold.nii.gz",
+    "bids_dataset/sub-02/ses-post/func/sub-02_ses-post_task-livingnonlivingdecisionwithplainormirrorreversedtext_run-02_events.tsv",
+    "bids_dataset/sub-02/ses-post/anat/sub-02_ses-post_inplaneT2.nii.gz",
+    "bids_dataset/sub-02/ses-post/anat/sub-02_ses-post_T1w.nii.gz",
+    "bids_dataset/sub-03/ses-pre/func/sub-03_ses-pre_task-livingnonlivingdecisionwithplainormirrorreversedtext_run-01_bold.nii.gz",
+    "bids_dataset/sub-03/ses-pre/func/sub-03_ses-pre_task-livingnonlivingdecisionwithplainormirrorreversedtext_run-01_events.tsv",
+    "bids_dataset/sub-03/ses-pre/func/sub-03_ses-pre_task-livingnonlivingdecisionwithplainormirrorreversedtext_run-02_bold.nii.gz",
+    "bids_dataset/sub-03/ses-pre/func/sub-03_ses-pre_task-livingnonlivingdecisionwithplainormirrorreversedtext_run-02_events.tsv",
+    "bids_dataset/sub-03/ses-pre/anat/sub-03_ses-pre_inplaneT2.nii.gz",
+    "bids_dataset/sub-03/ses-pre/anat/sub-03_ses-pre_T1w.nii.gz",
+    "bids_dataset/sub-03/ses-post/func/sub-03_ses-post_task-livingnonlivingdecisionwithplainormirrorreversedtext_run-01_bold.nii.gz",
+    "bids_dataset/sub-03/ses-post/func/sub-03_ses-post_task-livingnonlivingdecisionwithplainormirrorreversedtext_run-01_events.tsv",
+    "bids_dataset/sub-03/ses-post/func/sub-03_ses-post_task-livingnonlivingdecisionwithplainormirrorreversedtext_run-02_bold.nii.gz",
+    "bids_dataset/sub-03/ses-post/func/sub-03_ses-post_task-livingnonlivingdecisionwithplainormirrorreversedtext_run-02_events.tsv",
+    "bids_dataset/sub-03/ses-post/anat/sub-03_ses-post_inplaneT2.nii.gz",
+    "bids_dataset/sub-03/ses-post/anat/sub-03_ses-post_T1w.nii.gz"
+]
+
+def file_list_to_tree(file_list):
+    # Convert filelist to tree
+    dirTree = {}
+    for item in file_list:
+        path_parts = item.split('/')
+        sub_obj = dirTree
+        for part in path_parts:
+            if not part in sub_obj:
+                sub_obj[part] = {} if path_parts.index(part) < len(path_parts) - 1 else 'file'
+            sub_obj = sub_obj[part]
+
+    # Convert object lists to arrays
+    def objToArr (obj):
+        arr = []
+        for key in obj:
+            if obj[key] == 'file':
+                arr.append({'filename': key})
+            else:
+                arr.append({'name': key, 'children': objToArr(obj[key])})
+        return arr
+
+    return objToArr(dirTree)[0]
+
+
+def setup_download():
+    global session
+    session = requests.Session()
+    # all the requests will be performed as root
+    session.params = {
+        'user': 'test@user.com',
+        'root': True
+    }
+
+    # Convert file list to a tree
+    dataset = file_list_to_tree(file_list)
+
+    # Create a group
+    test_data.group_id = 'test_group_' + str(int(time.time()))
```

Comment:
please change this line to
`test_data.group_id = 'test_group_' + str(int(time.time()*1000))`
we then avoid to use the same group for two different test.